### PR TITLE
Week 3

### PR DESCRIPTION
**Pull request by Yusuf-ELSwaify**

### CODE REVIEW

File: week-3/leetcode/1832-check-if-the-sentence-is-pangram/Solution.java
Comment (mohamedshaker9) [inline week-3/leetcode/1832-check-if-the-sentence-is-pangram/Solution.java:1]: (Otional) 
try to solve this again with more readable and quickly understandable code than brilliant one but complex

File: week-3/introduction_to_java_book_tasks/_10_3_my_integer/src/main/java/job_hacker/tasks/models/MyInteger.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    }
+    public boolean isOdd(){
+        return (value & 1) == 1;
+    }
+    public boolean isPrime(){
+        if (value <= 1) {
+            return false;
+        }
+        if (value % 2 == 0) {
+            return value == 2;
```
Comment (mohamedshaker9) [inline week-3/introduction_to_java_book_tasks/_10_3_my_integer/src/main/java/job_hacker/tasks/models/MyInteger.java:23]: return true not 2

File: week-3/introduction_to_java_book_tasks/_10_3_my_integer/src/main/java/job_hacker/tasks/models/MyInteger.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        }
+        if (value % 3 == 0) {
+            return value == 3;
```
Comment (mohamedshaker9) [inline week-3/introduction_to_java_book_tasks/_10_3_my_integer/src/main/java/job_hacker/tasks/models/MyInteger.java:26]: return true

File: week-3/introduction_to_java_book_tasks/_11_1_Triangle_task/src/main/java/job_hacker/tasks/models/GeometricObject.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package job_hacker.tasks.models;
+
+import java.util.Date;
+
+public class GeometricObject {
+    String color;
+    boolean filled;
```
Comment (mohamedshaker9) [inline week-3/introduction_to_java_book_tasks/_11_1_Triangle_task/src/main/java/job_hacker/tasks/models/GeometricObject.java:7]: for good naming you could rename this to isFilled

File: week-3/introduction_to_java_book_tasks/_9_4_using_random/src/main/java/job_hacker/tasks/Main.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package job_hacker.tasks;
+
+import java.util.Arrays;
+import java.util.Random;
+
+public class Main {
+    public static void main(String[] args) {
+        Random rand = new Random(1000);
+        for (int i = 0;; i++){
```
Comment (mohamedshaker9) [inline week-3/introduction_to_java_book_tasks/_9_4_using_random/src/main/java/job_hacker/tasks/Main.java:9]: why made this please use for loop properly

**Review by mohamedshaker9 — CHANGES_REQUESTED** [review]

File: week-3/banking-app/src/main/java/job_hacker/example/models/accounts/Account.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
+package job_hacker.example.models.accounts;
+
+public class Account {
+    protected static int staticAccountNumber = 1;
```
[the remark this answers is not in the document]
Comment (Yusuf-ELSwaify) [reply week-3/banking-app/src/main/java/job_hacker/example/models/accounts/Account.java:4]: It's similar to the auto-increment primary key that the database generates by default, not the repository job, but here we don't have a DB.

File: week-3/introduction_to_java_book_tasks/_11_1_Triangle_task/src/main/java/job_hacker/tasks/models/GeometricObject.java (hunk 3 above)
Comment (Yusuf-ELSwaify) [reply week-3/introduction_to_java_book_tasks/_11_1_Triangle_task/src/main/java/job_hacker/tasks/models/GeometricObject.java:7]: Do you mean the getter? I called it like that by the way.
But that is not the naming convention for the field name itself, I guess.

File: week-3/introduction_to_java_book_tasks/_9_4_using_random/src/main/java/job_hacker/tasks/Main.java (hunk 4 above)
Comment (Yusuf-ELSwaify) [reply week-3/introduction_to_java_book_tasks/_9_4_using_random/src/main/java/job_hacker/tasks/Main.java:9]: I meant to use it like that to stop the last comma and not for vacuum.
I don't have another better solution.

File: week-3/banking-app/src/main/java/job_hacker/example/models/accounts/Account.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+            this.balance = balance;
+
+        ++staticAccountNumber;
+    }
+
+    public double deposit(double deposit) {
+        balance += deposit;
+        return balance;
+    }
+    public double withdraw(double money) {
```
[the remark this answers is not in the document]
Comment (Yusuf-ELSwaify) [reply week-3/banking-app/src/main/java/job_hacker/example/models/accounts/Account.java:24]: edited

File: week-3/banking-app/src/main/java/job_hacker/example/BankApplication.java
[the remark this answers is not in the document]
Comment (Yusuf-ELSwaify) [reply week-3/banking-app/src/main/java/job_hacker/example/BankApplication.java:1]: Done.

File: week-3/introduction_to_java_book_tasks/_10_3_my_integer/src/main/java/job_hacker/tasks/models/MyInteger.java (hunk 1 above)
Comment (Yusuf-ELSwaify) [reply week-3/introduction_to_java_book_tasks/_10_3_my_integer/src/main/java/job_hacker/tasks/models/MyInteger.java:23]: Why?
That will result in failure in many test cases.

File: week-3/banking-app/src/main/java/job_hacker/example/models/accounts/Account.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (mohamedshaker9) [reply week-3/banking-app/src/main/java/job_hacker/example/models/accounts/Account.java:4]: Yep, you are right but we think in coding with oop desgin best practices and clean coding.

File: week-3/introduction_to_java_book_tasks/_10_3_my_integer/src/main/java/job_hacker/tasks/models/MyInteger.java (hunk 1 above)
Comment (mohamedshaker9) [reply week-3/introduction_to_java_book_tasks/_10_3_my_integer/src/main/java/job_hacker/tasks/models/MyInteger.java:23]: Ok, you are right

File: week-3/introduction_to_java_book_tasks/_11_1_Triangle_task/src/main/java/job_hacker/tasks/models/GeometricObject.java (hunk 3 above)
Comment (mohamedshaker9) [reply week-3/introduction_to_java_book_tasks/_11_1_Triangle_task/src/main/java/job_hacker/tasks/models/GeometricObject.java:7]: For naming convention it recommended to start the naming with "is" for boolean variables and getter the same. The setter without is

**Pull request merged**